packages/protocol/test/tokenomics/utils.ts

```diff
@@ -67,4 +98,192 @@ const sendTinyEtherToZeroAddress = async (signer: any) => {
         .wait(1);
 };
 
-export { sendTinyEtherToZeroAddress, onNewL2Block };
+async function initTokenomicsFixture(mintTkoToProposer: boolean = true) {
+    const l1Provider = getL1Provider();
+
+    l1Provider.pollingInterval = 100;
+
+    const signers = await hardhatEthers.getSigners();
+    const l1Signer = signers[0];
+
+    const l2Provider = getL2Provider();
+
+    const l2Signer = await getDefaultL2Signer();
+
+    const l2AddressManager = await deployAddressManager(l2Signer);
+    const taikoL2 = await deployTaikoL2(l2Signer, l2AddressManager, false);
+
+    const genesisHash = taikoL2.deployTransaction.blockHash as string;
+    const genesisHeight = taikoL2.deployTransaction.blockNumber as number;
+
+    const l1AddressManager = await deployAddressManager(l1Signer);
+    const taikoL1 = await deployTaikoL1(
+        l1AddressManager,
+        genesisHash,
+        true,
+        defaultFeeBase
+    );
+    const { chainId } = await l1Provider.getNetwork();
+
+    const [proposerSigner, proverSigner] = await createAndSeedWallets(
+        2,
+        l1Signer
+    );
+
+    const tkoTokenL1 = await deployTkoToken(
+        l1Signer,
+        l1AddressManager,
+        taikoL1.address
+    );
+
+    await (
+        await l1AddressManager.setAddress(
+            `${chainId}.tko_token`,
+            tkoTokenL1.address
+        )
+    ).wait(1);
+
+    const { chainId: l2ChainId } = await l2Provider.getNetwork();
+
+    await (
+        await l1AddressManager.setAddress(`${l2ChainId}.taiko`, taikoL2.address)
+    ).wait(1);
+
+    await (
+        await l1AddressManager.setAddress(
+            `${chainId}.proof_verifier`,
+            taikoL1.address
+        )
+    ).wait(1);
+
+    if (mintTkoToProposer) {
+        const mintTx = await tkoTokenL1
+            .connect(l1Signer)
+            .mintAnyone(
+                await proposerSigner.getAddress(),
+                ethers.utils.parseEther("100")
+            );
+
+        await mintTx.wait(1);
+    }
+
+    // set up interval mining so we always get new blocks
+    await l2Provider.send("evm_setAutomine", [true]);
+
+    // send transactions to L1 so we always get new blocks
+    const interval = setInterval(
```

Comment:
Do u think we need to change the L1 test node to a hardhat node as well? and then we can simply use https://hardhat.org/hardhat-network/docs/reference#evm_setintervalmining here?